shpc/main/registry/remote.py

```diff
@@ -202,19 +211,32 @@ def _update_clone_cache(self):
         """
         logger.warning(
             "Remote %s is not deploying a Registry API, falling back to clone."
-            % self.source
+            % self.url
         )
         tmpdir = self.clone()
-        for dirname, module in self.iter_modules():
+        for module in Filesystem(tmpdir).iter_modules():
             # Minimum amount of metadata to function here
-            config_url = get_module_config_url(self.source, module)
+            config_url = self.get_module_config_url(module)[0]
             self._cache[module] = {
                 "config": shpc.utils.read_yaml(
-                    os.path.join(dirname, module, "container.yaml")
+                    os.path.join(tmpdir, module, "container.yaml")
                 ),
                 "config_url": config_url,
             }
-        shutil.rmtree(tmpdir)
+        self.cleanup()
+
+    def get_module_config_url(self, module_name):
+        """
+        Get the raw address of the config (container.yaml)
+        """
+        (_, _, domain, repo_path) = self.url.split("/", 3)
```

Comment:
Generally my preference is not the tuplized version, so e.g., without the extra () - there were a few instances of this added here!